test/e2e/single-node-3.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+set -e
```

Comment:
```suggestion
set -eu
```
Minor: I prefer also using `set -u` to error out on unset variables. If a variable can be unset wihtout it being an error you can explicitly state it with using the variable like this `${varname:-}`